tests/components/bayesian/test_binary_sensor.py

```diff
@@ -17,491 +16,488 @@
     STATE_UNKNOWN,
 )
 from homeassistant.core import Context, callback
-from homeassistant.setup import async_setup_component, setup_component
+from homeassistant.setup import async_setup_component
 
 from tests.async_mock import patch
-from tests.common import get_test_home_assistant
-
-
-class TestBayesianBinarySensor(unittest.TestCase):
-    """Test the threshold sensor."""
-
-    def setup_method(self, method):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-
-    def teardown_method(self, method):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    def test_load_values_when_added_to_hass(self):
-        """Test that sensor initializes with observations of relevant entities."""
-
-        config = {
-            "binary_sensor": {
-                "name": "Test_Binary",
-                "platform": "bayesian",
-                "observations": [
-                    {
-                        "platform": "state",
-                        "entity_id": "sensor.test_monitored",
-                        "to_state": "off",
-                        "prob_given_true": 0.8,
-                        "prob_given_false": 0.4,
-                    }
-                ],
-                "prior": 0.2,
-                "probability_threshold": 0.32,
-            }
+
+
+async def test_load_values_when_added_to_hass(hass):
+    """Test that sensor initializes with observations of relevant entities."""
+
+    config = {
+        "binary_sensor": {
+            "name": "Test_Binary",
+            "platform": "bayesian",
+            "observations": [
+                {
+                    "platform": "state",
+                    "entity_id": "sensor.test_monitored",
+                    "to_state": "off",
+                    "prob_given_true": 0.8,
+                    "prob_given_false": 0.4,
+                }
+            ],
+            "prior": 0.2,
+            "probability_threshold": 0.32,
         }
+    }
+
+    hass.states.async_set("sensor.test_monitored", "off")
+    await hass.async_block_till_done()
+
+    assert await async_setup_component(hass, "binary_sensor", config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)